python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -364,6 +364,8 @@ def _get_current_timestamp(self, current_time: Optional[datetime]) -> float:
         return current_time.timestamp()
 
     def get_num_partitions_in_window(self, time_window: TimeWindow) -> int:
+        if time_window.start >= time_window.end:
```

Comment:
datetime comparisons freak me out, maybe they shouldn't as much now that we are off of pendulum. But can this be time_window.start.timestamp() and time_window.end.timestamp() instead?